packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphCounterBlock.ts

```diff
@@ -0,0 +1,29 @@
+import type { FlowGraphContext } from "../../../flowGraphContext";
+import type { FlowGraphDataConnection } from "../../../flowGraphDataConnection";
+import { RichTypeNumber } from "../../../flowGraphRichTypes";
+import type { FlowGraphSignalConnection } from "../../../flowGraphSignalConnection";
+import { FlowGraphWithOnDoneExecutionBlock } from "../../../flowGraphWithOnDoneExecutionBlock";
+
+export class FlowGraphCounterBlock extends FlowGraphWithOnDoneExecutionBlock {

```

Comment:
I know it is a pain but I recommend to document now instead of having a monster pass at the end ;)

Reply:
@RaananW should we capture it at build time?

Reply:
yes. I just need to find a few free hours to finalize the implementation

Reply:
I had the comments but forgot to commit 🤦 thanks yall for catching